vexim/sitechangesubmit.php

```diff
@@ -1,99 +1,145 @@
-        maxmsgsize=:maxmsgsize, pipe=:pipe, max_accounts=:max_accounts,
-        quotas=:quotas, sa_tag=:sa_tag, sa_refuse=:sa_refuse,
-        spamassassin=:spamassassin, enabled=:enabled
-        WHERE domain_id=:domain_id";
-      $sth = $dbh->prepare($query);
-      $success = $sth->execute(array(':uid'=>$uid, ':gid'=>$gid,
-        ':avscan'=>$_POST['avscan'], ':maxmsgsize'=>$_POST['maxmsgsize'],
-        ':pipe'=>$_POST['pipe'], ':max_accounts'=>$_POST['max_accounts'],
-        ':quotas'=>$_POST['quotas'],
-        ':sa_tag'=>((isset($_POST['sa_tag'])) ? $_POST['sa_tag'] : 0),
-        ':sa_refuse'=>((isset($_POST['sa_refuse'])) ? $_POST['sa_refuse'] : 0),
-        ':spamassassin'=>$_POST['spamassassin'], ':enabled'=>$_POST['enabled'],
-        ':domain_id'=>$_POST['domain_id'],
-        ));
-  }
+    // User can specify either UID, or username, the former being preferred.
+    // Using posix_getpwuid/posix_getgrgid even when we have an UID is so we
+    // are sure the UID exists.
+    if (isset ($_POST['uid'])) {$uid = $_POST['uid'];}
+    if (isset ($_POST['gid'])) {$gid = $_POST['gid'];}
+
+    if ($userinfo = @posix_getpwuid($uid)) {
+      $uid = $userinfo['uid'];
+    } elseif ($userinfo = @posix_getpwnam($uid)) {
+      $uid = $userinfo['uid'];
+    } else {
+      header("Location: site.php?failuidguid={$_POST['domain']}");
+      die;
+    }
 
-  if ($success) {
-    header ("Location: site.php?updated={$_POST['domain']}");
-    die;
-  }
+    if ($groupinfo = @posix_getgrgid($gid)) {
+      $gid = $groupinfo['gid'];
+    } elseif ($groupinfo = @posix_getgrnam($gid)) {
+      $gid = $groupinfo['gid'];
+    } else {
+      header("Location: site.php?failuidguid={$_POST['domain']}");
+      die;
+    }
+    $query = "UPDATE domains SET uid=:uid, gid=:gid, avscan=:avscan,
+            maxmsgsize=:maxmsgsize, pipe=:pipe, max_accounts=:max_accounts,
+            quotas=:quotas, sa_tag=:sa_tag, sa_refuse=:sa_refuse,
+            spamassassin=:spamassassin, enabled=:enabled
+            WHERE domain_id=:domain_id";
+    $sth = $dbh->prepare($query);
+    $success = $sth->execute(array(':uid' => $uid, ':gid' => $gid,
+        ':avscan' => $_POST['avscan'], ':maxmsgsize' => $_POST['maxmsgsize'],
+        ':pipe' => $_POST['pipe'], ':max_accounts' => $_POST['max_accounts'],
+        ':quotas' => $_POST['quotas'],
+        ':sa_tag' => ((isset($_POST['sa_tag'])) ? $_POST['sa_tag'] : 0),
+        ':sa_refuse' => ((isset($_POST['sa_refuse'])) ? $_POST['sa_refuse'] : 0),
+        ':spamassassin' => $_POST['spamassassin'], ':enabled' => $_POST['enabled'],
+        ':domain_id' => $_POST['domain_id'],
+    ));
+}
 
-# Just-in-case catchall
-header ("Location: site.php?failupdated={$_POST['domain']}");
-?>
-<!-- Layout and CSS tricks obtained from http://www.bluerobot.com/web/layouts/ -->
+if ($success) {
+  header("Location: site.php?updated={$_POST['domain']}");
+  die;
+} else {
```

Comment:
No need for the `else` condition here, since we won't reach it anyway if the `if` is satisfied. Just leave the comment and the `header()` call. Although, while here, why not change the comment style to a more common `// `.